src/psyclone/tests/psyGen_test.py

```diff
@@ -398,6 +399,38 @@ def test_reset():
 # tests for class Call
 
 
+def test_invokes_can_always_be_printed():
+    '''Test that an Invoke instance can always be printed (i.e. is
+    initialised fully)'''
+    inv = Invoke(None, None, None)
+    assert inv.__str__() == "invoke()"
+
+
+def test_invokes_can_always_be_printed2():
+    '''Test that an Invoke instance can always be printed (i.e. is
+    initialised fully) 2'''
+
+    invoke_call = InvokeCall([], "TestName")
+    inv = Invoke(invoke_call, 12, DynSchedule)
+    # Name is converted to lower case if set in constructor of InvokeCall:
+    assert inv.__str__() == "invoke_testname()"
+
+    invoke_call._name = None
+    inv = Invoke(invoke_call, 12, DynSchedule)
+    assert inv.__str__() == "invoke_12()"
+
+    # Last test case: one kernel call - to avoid constructing
+    # the InvokeCall, parse an existing Fortran file"
+
+    _, invoke = parse(
+        os.path.join(BASE_PATH, "1.12_single_invoke_deref_name_clash.f90"),
+        api="dynamo0.3")
+
+    alg_invocation = invoke.calls.values()[0]
+    inv = Invoke(alg_invocation, 0, DynSchedule)
+    assert inv.__str__() == "invoke_0_testkern_type(a, f1_my_field, f1%my_field, m1, m2)"
+
+
```

Comment:
Tests are good although might as well be merged into a single test unless there's a good reason for having '1' and '2' separate. Line 431 is not pep8-compliant.

Reply:
Also, to keep pylint happy please add `  # pylint: disable=invalid-name` after the test name.